### PR TITLE
Add resource name kinds to the registry.

### PR DESCRIPTION
**Pull request by echlebek**
Signed-off-by: Eric Chlebek <eric@sensu.io>

### CODE REVIEW

File: internal/registry/registry.go [hunk 1]
```diff
@@ -22,29 +22,42 @@ const templateText = `package registry
...
+	metav1.TypeMeta{APIVersion: "{{ $t.APIVersion }}", Kind: "{{ lower $t.Kind }}"}: {{ replace $t.APIVersion "/" "" 1 }}.{{ $t.Kind }}{}, {{ end }}
+}
+
+func init() {
+	for k, v := range typeRegistry {
+		r, ok := v.(interface{ResourceName() string})
+		if ok {
+			newKey := metav1.TypeMeta{
+				APIVersion: k.APIVersion,
+				Kind: r.ResourceName(),
```
Comment (ccressent) [inline internal/registry/registry.go]: Should we also change the resource name tags as currently defined?

Right now for, say, `ObjectMeta`, the resource name is `objectMetas`. 

Don't we want to end up with `objectmetas` and `ObjectMetas` in the registry instead (or at least the all lowercase version)?

Comment (echlebek) [reply internal/registry/registry.go]: No, because:

```
apiGroup: core/v1
kind: check
```

is desirable in yaml configuration.

Comment (ccressent) [reply internal/registry/registry.go]: I'm sorry, I don't understand.

Based on the content of [resource.go for RBAC](https://github.com/sensu/sensu-go/blob/devel/new-api/apis/rbac/v1alpha1/resource.go), what I'm saying is that this loop effectively adds entries of this form, eg for `ClusterRole`:
```
metav1.TypeMeta{APIVersion: "rbac", Kind: "clusterRoles"}: rbac.ClusterRole{},
```
while we want an entry of this form:
```
metav1.TypeMeta{APIVersion: "rbac", Kind: "clusterroles"}: rbac.ClusterRole{},
```
That is, all lowercase and pluralized. It's fine for `Role` and `Subject`, but not the other types.

Comment (echlebek) [reply internal/registry/registry.go]: I see. I've amended the PR to change the resource name declaration in the RBAC internal API.

Comment (palourde) [reply internal/registry/registry.go]: While we are here, would it be hard to have `clusterroles` return `[]rbac.ClusterRole{}`?

Comment (echlebek) [reply internal/registry/registry.go]: Interesting. I'll look into it. :)

**Comment by palourde** [comment]
Let's merge that for now :)

**Pull request merged**